CMakeLists.txt

```
@@ -20,7 +20,8 @@ get_filename_component(include_dir     ${root_dir}/include     ABSOLUTE)
 get_filename_component(test_dir        ${root_dir}/test        ABSOLUTE)
 get_filename_component(example_dir     ${root_dir}/example     ABSOLUTE)
 
-ctulu_create_target(WaitFreeHashMapLibrary "WaitFreeHashMap" INTERFACE_INCLUDES ${include_dir} CXX 17 INTERFACE)
+# set parameter INTERFACE since the library is header only.
+ctulu_create_target(WaitFreeHashMapLibrary "WaitFreeHashMap" INTERFACE_INCLUDES SYSTEM ${include_dir} CXX 17 INTERFACE)
```

It's cool for end users but we need something to get warnings during development as well.

---

Yes indeed, what do you think of this:
```suggestion
if(CMAKE_BUILD_TYPE STREQUAL "Debug")
    message(STATUS "Activating warning on library")
    ctulu_create_target(WaitFreeHashMapLibrary "WaitFreeHashMap" INTERFACE_INCLUDES ${include_dir} CXX 17 INTERFACE)
else()
    ctulu_create_target(WaitFreeHashMapLibrary "WaitFreeHashMap" INTERFACE_INCLUDES SYSTEM ${include_dir} CXX 17 INTERFACE)
endif()
```

---

I'm OK with it :+1:

---

Merged